src/python/pants/backend/native/subsystems/binaries/gcc.py

```diff
@@ -40,10 +40,10 @@ def path_entries(self):
 
   @memoized_method
   def _common_lib_dirs(self, platform):
-    lib64_tuples = platform.resolve_for_enum_variant({
-      'darwin': [],
-      'linux': [('lib64',)],
-    })
+    lib64_tuples = {
+      Platform.darwin: [],
+      Platform.linux: [('lib64',)]
+    }[platform]
```

Comment:
So kind of like you can see with the `_AlgebraicDataclass` mixin introduced in #8437, while we can revert to mypy-compatible typing, we can move the extensions made *on top of* more standard python types into mixins (e.g. `pants.util.objects.DatatypeMixin` or `ChoicesMixin`. One of the things that `.resolve_for_enum_variant()` has that this still loses is exhaustiveness checking -- as in, that *only* the correct keys are provided, and that *all* cases are accounted for. It also means we *don't* have to ever make these types of refactorings if python 4 introduces a new type of dataclasses, for example (**stable API!!!**). Could we consider keeping the `ChoicesMixin`, specializing it into a `StdlibEnumWrapperMixin` for stdlib enums to implement `.resolve_for_enum_variant()`, and then having all the new stdlib `Enum` classes also mix that in, and then possibly avoid having to make these changes?